pkg/storage/store_pool.go

```diff
@@ -327,6 +327,29 @@ func (sp *StorePool) deadReplicasGossipUpdate(_ string, content roachpb.Value) {
 	detail.deadReplicas = deadReplicas
 }
 
+// updateLocalStoreAfterRebalance is used to update local copy of target store
+// after we make an rebalance immediately.
+func (sp *StorePool) updateLocalStoreAfterRebalance(
+	storeID roachpb.StoreID, repl *Replica, changeType roachpb.ReplicaChangeType,
+) {
+	sp.detailsMu.Lock()
+	defer sp.detailsMu.Unlock()
+	detail := *sp.getStoreDetailLocked(storeID)
+	switch changeType {
+	case roachpb.ADD_REPLICA:
+		detail.desc.Capacity.LogicalBytes += repl.GetMVCCStats().Total()
+		if qps, dur := repl.writeStats.avgQPS(); dur >= MinStatsDuration {
+			detail.desc.Capacity.WritesPerSecond += qps
+		}
+	case roachpb.REMOVE_REPLICA:
+		detail.desc.Capacity.LogicalBytes -= repl.GetMVCCStats().Total()
```

Comment:
Nit, but it'd be nice to make sure we aren't reducing these values to less than 0 here. It's very unlikely, but is possible for us to have already received updated gossip from the store which would technically make it possible for subtracting these values to end up at negative values.

Reply:
Done.